x-pack/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -247,38 +248,45 @@ export const createCustomThresholdExecutor = ({
           return acc;
         }, []);
 
-        const alert = alertFactory(
-          `${group}`,
-          reason,
-          actionGroupId,
-          additionalContext,
-          evaluationValues,
-          groupByKeysObjectMapping[group]
-        );
-        const alertUuid = getAlertUuid(group);
-        const indexedStartedAt = getAlertStartedDate(group) ?? startedAt.toISOString();
-        scheduledActionsCount++;
+        try {
+          const alert = alertFactory(
+            `${group}`,
+            reason,
+            actionGroupId,
+            additionalContext,
+            evaluationValues,
+            groupByKeysObjectMapping[group]
+          );
+          const alertUuid = getAlertUuid(group);
+          const indexedStartedAt = getAlertStartedDate(group) ?? startedAt.toISOString();
+          scheduledActionsCount++;
 
-        alert.scheduleActions(actionGroupId, {
-          alertDetailsUrl: await getAlertUrl(
-            alertUuid,
-            spaceId,
-            indexedStartedAt,
-            alertsLocator,
-            basePath.publicBaseUrl
-          ),
-          group: groupByKeysObjectMapping[group],
-          reason,
-          timestamp,
-          value: alertResults.map((result, index) => {
-            const evaluation = result[group];
-            if (!evaluation) {
-              return null;
-            }
-            return formatAlertResult(evaluation).currentValue;
-          }),
-          ...additionalContext,
-        });
+          alert.scheduleActions(actionGroupId, {
+            alertDetailsUrl: await getAlertUrl(
+              alertUuid,
+              spaceId,
+              indexedStartedAt,
+              alertsLocator,
+              basePath.publicBaseUrl
+            ),
+            group: groupByKeysObjectMapping[group],
+            reason,
+            timestamp,
+            value: alertResults.map((result, index) => {
+              const evaluation = result[group];
+              if (!evaluation) {
+                return null;
+              }
+              return formatAlertResult(evaluation).currentValue;
+            }),
+            ...additionalContext,
+          });
+        } catch (err) {
```

Comment:
Ideally, we would like for the rule executors to respect the alert limit instead of throwing an error in the executor. When an executor error is thrown, the rest of framework execution is short circuited (so the actions that were scheduled will not be executed and alerts from this rule execution won't be processed). Here is the RFC for the max alerts circuit breaker for more details: https://docs.google.com/document/d/1u9U8fB2QYUhfxqwhSX1PRxnn4QofvqfA8fsnOVEIwIY/edit#heading=h.2b1v1tr0ep8m

Here is the PR where we added the limit check to the stack rules: https://github.com/elastic/kibana/pull/141000

Essentially you will want to access the configured limit using

```
const alertLimit = alertFactory.alertLimit.getValue();
```

and then inside the executor, keep track of the number of alerts you've reported so you can report up to (and not exceeding) the limit. If at that point you know you have more alerts to report, you'll want to set `alertFactory.alertLimit.setLimitReached(true)`. This allows the framework to put the execution into a warning state (this should show up in the exec log and in the UI) but continue processing the alerts and triggering notifications. LMK if you have questions

Reply:
thanks for the info, i will take care of it.

Reply:
updated based on above comment